lib/matplotlib/axis.py

```diff
@@ -1339,22 +1374,15 @@ def get_major_ticks(self, numticks=None):
             numticks = len(self.get_major_locator()())
         if len(self.majorTicks) < numticks:
             # update the new tick label properties from the old
+            protoTick = self.majorTicks[0]
             for i in range(numticks - len(self.majorTicks)):
```

Comment:
could even just replace the outer "if len(self.majorTicks) < numticks" by "while len() < numticks" and remove one indent level.